### PR TITLE
Support browser dist without canvas dependency

### PR DESCRIPTION
**Pull request by bz2**
Thanks for picking this project up again and working on a new beta release! We're using canvg just for rending of svgs to canvas on IE11, and not the server side rendering support. So, for us the `1.5` branch is fine, but updating to `2.0.0-beta.0` will not import without the `canvas` dep, which is not actually used by the browser file. This change proposes one possible fix, am happy to make changes if there are better ideas.

In a non-node environment, none of `jsdom`, `xmldom`, or `canvas` packages are required, instead browser primitives are used. Make the handling of these three requirements consistent, omit them from the browser build, and mark dependencies peer plus dev rather than direct.

Update dist files, and package lock to latest minor versions.

Is there better way of handling these sometimes-of optional package deps with npm, than marking them as peers and allowing the requirement to not be satisfied in some cases?

### CODE REVIEW

**Comment by gabelerner** [comment]
@bz2 thanks for the PR! so since canvas is natively support on browser and therefore a peer dependency, you went ahead and added xmldom/jsdom as peer dependencies as well because they're also natively supported? If so, I agree and it seems like an oversight that should be in master. I'll go ahead and test both environments and merge and pop the beta version. Thanks!

**Comment by bz2** [comment]
@gabelerner Thanks for reviewing! Yes, this just aims to make the handling of the browser supplied features consistent. Will need a note in the changelog to install the additional peer deps for the node case, and that the complaint about missing deps for the browser case can be safely ignored.

Only other option I've come up with is a package split, which would likely be more difficult to maintain right now.

**Comment by gabelerner** [comment]
@bz2 i guess i have a follow up question, what do you mean `will not import without the canvas dep`? how is the 2.0 breaking in your flow? sure there are other files installed (xmldom/jsdom/etc) but they should be ignored and the browser should point to https://github.com/canvg/canvg/blob/master/package.json#L26

**Comment by bz2** [comment]
Okay, so the clarify. At the moment if you have a project that depends on canvg 2.0 beta, and only builds in a browser context, on install *without* [canvas](https://github.com/Automattic/node-canvas/) npm warns but succeeds:
```
npm WARN canvg@2.0.0-beta.0 requires a peer of canvas@^2.1.0 but none is installed. You must install peer dependencies yourself.
```
But build will fail on trying to import canvg:
```
ERROR in ./node_modules/canvg/dist/browser/canvg.min.js
Module not found: Error: Can't resolve 'canvas' in '.../client/node_modules/canvg/dist/browser'
 @ ./node_modules/canvg/dist/browser/canvg.min.js 1:133-150
```

On install *with* canvas, that happens:
```
> canvas@2.3.1 install .../client/node_modules/canvas
> node-pre-gyp install --fallback-to-build

node-pre-gyp WARN Using request for node-pre-gyp https download
[canvas] Success: ".../client/node_modules/canvas/build/Release/canvas.node" is installed via remote
```
Which takes a long time because it's downloading node binaries over http, and on build what actually gets included in the browser bundle is this stub file:
https://github.com/Automattic/node-canvas/blob/master/browser.js

So, it functions, but requires binary downloads or c build which ends up not being used.

With this branch, the second case is the same (but requires node projects to also include jsdom and xmldom directly in their dependencies alongside canvg as was as canvas), but the first case successfully builds, and additionally does not install jsdom or xmldom unless they are used anyway.

**Comment by gabelerner** [comment]
yea makes sense, i'll probably merge these changes and update the readme a bit with more instructions requiring node users to install these packages.

**Comment by bz2** [comment]
I can update the readme and changelog in this branch after rebasing.

See also https://github.com/npm/npm/issues/14185 for the (old) upstream npm issue about optional deps.

**Comment by bz2** [comment]
@gabelerner Rebased and added docs.

**Comment by gabelerner** [comment]
this looks good, gonna test it locally then merge.

i guess it's primarily meant to be a browser tool, since those are the ones that lack svg libs. server side, you have plenty of other svg libs you can use (i can't really believe mine is the best here, something like a headless chrome is probably better) so i'm happy to have the browser path be the path of least resistance.

File: dist/node/canvg.d.ts [hunk 1]
```diff
@@ -1,23 +0,0 @@
-declare namespace canvg {
```
Comment (gabelerner) [inline dist/node/canvg.d.ts]: any reason the type file was deleted?

Comment (bz2) [reply dist/node/canvg.d.ts]: Fixed.

Am used to `rm -rf dist/ && npm run build` being a sane thing to do, didn't realise there were precious files under there.

**Comment by gabelerner** [comment]
@bz2 thanks for this! i'll push out a new beta and then let it simmer for a couple weeks and then thoughts on making it an official 2.0 release?

**Comment by bz2** [comment]
@gabelerner Sounds good to me. Can update our app to the new beta and pass along feedback prior to final release.

**Pull request merged**